modules/library/main/src/main/java/org/geotools/filter/function/StandardDeviationFunction.java

```diff
@@ -96,4 +111,41 @@ private Double getMax(int index, int numClasses, double average, double standard
         if (index < 0 || index >= numClasses - 1) return null;
         return Double.valueOf(average - (((numClasses / 2.0) - 1 - index) * standardDeviation));
     }
+
+    private double[] getPercentages(
+            FeatureCollection features,
+            RangedClassifier classifier,
+            Expression attr,
+            double standardDeviation)
+            throws IOException {
+        int classSize = classifier.getSize();
+        Object firstMax = classifier.getMax(0);
+        Filter less = FF.less(attr, FF.literal(firstMax));
+        int sizeFirstClass = features.subCollection(less).size();
+        int totalSize = features.size();
+        double[] percentages = new double[classSize];
+        // we don't know the min value in the collection because the
+        // the first interval is open to infinity to the left.
```

Comment:
Thanks @aaime I've added a check on percentages one by one. The openess to infinity of the last class is handled here https://github.com/geotools/geotools/pull/3078/commits/9678b0934a080c3a4e343c6a456ff193a4efef87#diff-2319f1367ac925907808cd3d8f0fa864R283 where the index returned from the query is checked on the percentages length and in case  the resulting percentage is merged on the one of the last class. I've added also a test to check this behaviour.